### PR TITLE
Add black and github workflow rework

### PR DESCRIPTION
**Pull request by alinabuzachis**
##### SUMMARY

Add black and github workflow rework

##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: .github/workflows/black.yml
Comment (GomathiselviS) [inline .github/workflows/black.yml:1]: The linters job in changelog_and linters.yml will take care of 'black' check if 'black' is added to tox.ini. I do not think a separate workflow is needed.

Comment (alinabuzachis) [reply .github/workflows/black.yml:1]: black should be run against main only, so I removed black from top and re-added the workflow for black.

Comment (tremble) [reply .github/workflows/black.yml:1]: tox might be the better option here.  We can add it to main's tox and as long as we don't backport the tox update then we'll only apply this *style* change to 6.0.0 and later.

**Review by tremble — COMMENTED** [review]
Does tests.yml need to be removed?  This looks like the unit/sanity tests are duplicated now

**Comment by alinabuzachis** [comment]
> Does tests.yml need to be removed? This looks like the unit/sanity tests are duplicated now

Yes, thanks!

**Review by GomathiselviS — APPROVED** [review]

**Pull request merged**